apps/dashr-oil-and-gas-ternary/app.R

```diff
@@ -0,0 +1,761 @@
+                    'min'= -2, 'linewidth'= 1.5, 'ticks'= 'outside')),
+    "margin" = list(l=60, r=0, t=0, b=0),
+    "paper_bgcolor" = "#192444",
+    "plot_bgcolor" = "#192444",
+    "showLegend" = FALSE,
+    "font" = list("color" = "white"),
+    "annotations" = list("visible" = FALSE),
+    "autosize" = TRUE
+  ) 
+  hovertemplate = "<b> %{text}</b><br><br> Quartz: %{a:.0f}<br>Carbonate : 
+  %{b:.0f}<br> Clay: %{c:.0f}<extra></extra>"
+  
+  formations = list(unique(dff['fm_name']))
+  data_traces = list()
+  j = 1
+  for (key in formations[[1]]$fm_name) {
+    if(length(selected_data) > 0){
+      select_indices = selected_data[[key]]
+    } else{
+      select_indices = NULL
+    }
+    data_traces[[j]] = list(
+      text=as.list(paste('Well ID', 
+                         dff[dff$fm_name == formation,'RecordNumber'])),
+      name=key,
+      customdata=dff[dff$fm_name == key,'RecordNumber'],
+      type='scatterternary',
+      a=dff[dff$fm_name == key,'Quartz'],
+      b=dff[dff$fm_name == key,'Carbonate'],
+      c=dff[dff$fm_name == key,'Clay'],
+      mode='markers',
+      hovertemplate = hovertemplate,
+      showlegend=TRUE,
+      marker=list('color' = colormap[[key]], 'size' = 8, 'line' = list('color' = '#000000', 'width'= 0.2)),
+      selectedpoints=select_indices,
+      visible=marker_visible
+    )
+    #data_traces = append(data_traces,new_data_trace)
+    j = j+1
+  }
+  FData = append(contour_traces, contour_text)
+  FData = append(FData, data_traces)
+  return(list('data' = FData, 'layout' = layout))
+}
+
+generate_contour_text_layer = function(contour_visible){
+  layer = list()
+  j = 1
+  for (key in labels(ternary_contour)) {
+    a= mean(unlist(lapply(ternary_contour[[key]], function(k){k[1]})))
+    b = mean(unlist(lapply(ternary_contour[[key]], function(k){k[2]})))
+    c = mean(unlist(lapply(ternary_contour[[key]], function(k){k[3]})))
+    key_br = gsub(" ", '<br>', key)
+    new_trace = generate_contour_text(a, b, c, key, key_br, contour_visible)
+    layer[[j]] = new_trace
+    j = j+1
+  }
+  return(layer)
+}
+
+generate_contour_text = function(a, b, c, name, text, visible){
+  return(list(
+    type='scatterternary',
+    a=list(a),
+    b=list(b),
+    c=list(c),
+    name=name,
+    text=list(text),
+    mode='text',
+    hoverinfo='all',
+    textposition='middle center',
+    textfont=list('size' = 11, 'color' = '#000000', 'family' = 'sans-serif'),
+    showlegend=FALSE,
+    legendgroup='Rock type',
+    visible=visible
+  ))
+}
+
+generate_formation_bar = function(dff, selected_data){
+  
+  formations = list(unique(dff['fm_name']))
+  xform <- list(categoryorder = "array", 
+                categoryarray = formations[[1]]$fm_name,
+                tickangle=-45, title="Formations")
+  data = list()
+  countt = list()
+  color = list()
+  
+  for (i in formations[[1]]$fm_name) {
+    counttt = nrow(dff[dff$fm_name == i,])
```

Comment:
Same with `counttt` here, and any other variation in which ≥ 2 `t`'s are added 🙂